cpp/test/Ice/timeout/AllTests.cpp

```diff
@@ -165,19 +165,25 @@ allTestsWithController(Test::TestHelper* helper, const ControllerPrx& controller
         {
             test(false);
         }
-        while (true)
+
+        try
         {
-            try
-            {
-                connection->getInfo();
-                this_thread::sleep_for(chrono::milliseconds(10));
-            }
-            catch (const Ice::ConnectionClosedException& ex)
-            {
-                // Expected.
-                test(ex.closedByApplication());
-                break;
-            }
+            closeFuture.get(); // Wait for close to complete.
+            test(false);
+        }
+        catch (const Ice::CloseTimeoutException&)
+        {
+            // Expected.
+        }
+
+        try
+        {
+            connection->getInfo();
+            test(false);
+        }
+        catch (const Ice::CloseTimeoutException& ex)
```

Comment:
Since we don't test `ex.closedByApplication()` anymore:
```suggestion
        catch (const Ice::CloseTimeoutException&)
```